daemon/api_notices.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package daemon
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/snapcore/snapd/overlord/auth"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/strutil"
+)
+
+var (
+	noticesCmd = &Command{
+		Path:       "/v2/notices",
+		GET:        getNotices,
+		ReadAccess: openAccess{},
+	}
+
+	noticeCmd = &Command{
+		Path:       "/v2/notices/{id}",
+		GET:        getNotice,
+		ReadAccess: openAccess{},
```

Comment:
until we have the distinction of public|private coming together with the per-user work we probably want authenticatedAccess here (that also matches UserOK on the pebble side)